rfcs/text/testing-refinement.md

```diff
@@ -0,0 +1,110 @@
+
+# Alternatives
+
+There is some duplication between the Cypress regression tests and the unit tests. Indeed, the act of setting knobs to various states to set a prerequisite state and the result asserted on is already done in many of the unit tests. An exercise to remove duplicate tests in the regression suite could be carried out. The recent removal of support of the ‘classic’ theme and related feature files reduced the regression suite run time by 1-1.5 hours. We could upgrade our Cypress package to a paid for service which would allow the tests to be parallelised rather than them all running in one thread, reducing the run time further. All of these actions though would not reduce the regression suite run time by the required factor nor provide the visual comparison testing capability that Applitools integration would.
+
+One advantage of integrating Applitools is having the option of rendering the DOM snapshots in many other operating system, browser, device and viewport configurations we like by using their Ultrafast Grid. This does not mean though that we are testing our components in these other configurations at the time of test execution. There are other vendors, including CrossBrowserTesting and BrowserStack, which would allow us to run our required tests directly in these other configurations of real machines. However, this would mean a change of automation framework from Cypress to Selenium. This seems like a drastic change at this point as all the files, including the support files like helper files, indexes, locators etc, would need rewriting. In addition, not all of these other products are free for open source repositories as Applitools is and don’t all provide a dashboard which is as user-friendly as that of Applitools.
+
+# Adoption strategy
+
+Developers would need to amend how they write the stories. As many examples as possible of various main configurations of the components will need to be included in one story without any knobs. Other React libraries such as Material-UI (https://material-ui.com) use this method of demonstrating their components. The current standard of documentation in the `Docs` tab will need to be maintained to provide a thorough description of how the component is meant to be used.
+
+QA Engineers will amend existing specs to leave just the steps to capture snapshots with Applitools. QA Engineers will need to identify additional scenarios and use cases not included in the main `Design System` directory as currently. These would be added to the `Test` directory, although it would be beneficial if QA Engineers were able to add these themselves with the help of Developers if required.
```

Comment:
It would be best to prefer unit tests, adding stories only when we want to do a visual comparison.

Reply:
I've added something to that effect in the newer version I've pushed for review again.